### PR TITLE
first distributor implementation

### PR DESCRIPTION
**Pull request by aschieb**

### CODE REVIEW

**Comment by ohadlevy** [comment]
awesome seeing this :100:

File: plugins/pulp_deb/plugins/distributors/steps.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 from gettext import gettext as _
 import logging
 import os
+import commands
```
Comment (barnabycourt) [inline plugins/pulp_deb/plugins/distributors/steps.py]: commands has been deprecated.  We should use subprocess instead

File: plugins/pulp_deb/plugins/distributors/steps.py [hunk 2]
```diff
@@ -58,6 +61,13 @@ def process_main(self):
         Publish all the deb metadata or create a blank deb if this has never been synced
         """
         # Write out repo metadata into the working directory
+        packfile = os.path.join(self.working_directory, "Packages")
+        status, out = commands.getstatusoutput("cd " + self.working_directory + "; dpkg-scanpackages -m . > " + packfile)
```
Comment (barnabycourt) [inline plugins/pulp_deb/plugins/distributors/steps.py]: other than using the commands library this line is also very long.  Did you use the pulp_deb/run_tests.py script to run the unit tests style guide checks?

File: plugins/pulp_deb/plugins/distributors/steps.py [hunk 3]
```diff
@@ -71,10 +81,15 @@ def __init__(self, **kwargs):
         self.redirect_context = None
         self.description = _('Publishing Deb Content.')
         self.unit = None
+        self.working_directory = kwargs["working_dir"]
```
Comment (barnabycourt) [inline plugins/pulp_deb/plugins/distributors/steps.py]: This line is not necessary

File: plugins/pulp_deb/plugins/distributors/steps.py [hunk 4]
```diff
@@ -71,10 +81,15 @@ def __init__(self, **kwargs):
...
+        os.makedirs(self.working_directory)
```
Comment (barnabycourt) [inline plugins/pulp_deb/plugins/distributors/steps.py]: This can also be removed

File: plugins/pulp_deb/plugins/distributors/steps.py [hunk 5]
```diff
@@ -71,10 +81,15 @@ def __init__(self, **kwargs):
...
 
     def process_main(self):
         """
         Publish all the deb files themselves
         """
-        # Symlink all sub directories of the storage dir except the refs directory
-        # Perhaps we should consider using PluginStepIterativeProcessingMixin
+        units = self.get_conduit().get_units()
+        for unit in units:
+            path = os.path.join(self.working_directory, os.path.basename(unit.storage_path))
```
Comment (barnabycourt) [inline plugins/pulp_deb/plugins/distributors/steps.py]: use self.get_working_dir() to get the current working directory

**Comment by barnabycourt** [comment]
Hi, Great to see the first submission for the new pulp_deb project.  I'd be happy to work with you on getting this in.  

The major items that need to be updated for this PR is the unit test coverage & the use of the commands module.  See the inline comments for details on those.  The major way to check all of that is to run the 'run-tests.py' script 

One other thing you might be interested in, I am starting to think through how we could support the more complicated nesting of repositories that Debian supports.  See https://pulp.plan.io/issues/80 for the story where that work is being tracked.

**Comment by mmoll** [comment]
@barnabycourt what's the status of this PR? AFAICT the comments of the first review were addressed.

File: plugins/pulp_deb/plugins/distributors/steps.py [hunk 6]
```diff
@@ -58,13 +60,25 @@ def process_main(self):
         Publish all the deb metadata or create a blank deb if this has never been synced
         """
         # Write out repo metadata into the working directory
+        packfile = os.path.join(self.get_working_dir(), "Packages")
+        dpkg_out = open(packfile, 'w')
+        proc = subprocess.Popen(['dpkg-scanpackages', '-m', '.'], cwd=self.get_working_dir(),
+                                stdout=subprocess.PIPE)
+        (out, err) = proc.communicate()
```
Comment (barnabycourt) [inline plugins/pulp_deb/plugins/distributors/steps.py]: out and err are never checked

**Comment by barnabycourt** [comment]
This PR is missing unit tests.

**Comment by barnabycourt** [comment]
@aschieb I merged this down and submitted a separate PR for it: https://github.com/pulp/pulp_deb/pull/7

**Comment by barnabycourt** [comment]
@aschieb This code was merged as part of PR #7

**Pull request closed without merging**